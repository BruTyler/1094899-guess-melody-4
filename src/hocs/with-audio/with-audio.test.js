import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withAudio from './with-audio.jsx';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

describe(`withAudio render suit`, () => {
  it(`withAudio render case`, () => {
    const generatedTree = renderer.create(
        <MockComponentWrapped
          src={``}
          isPlaying={false}
          onPlayButtonClick={() => {}}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});
