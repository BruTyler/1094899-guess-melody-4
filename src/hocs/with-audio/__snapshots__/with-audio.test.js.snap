// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudio render suit withAudio render case 1`] = `
<div>
  <audio />
</div>
`;
